update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java

```diff
@@ -0,0 +1,341 @@
+    private static final String INVALID_USERNAME = "invalidUsername";
+    public static final String NULL_STRING_REPRESENTATION = "null";
+
+    private static final String USERNAME_NOT_IMPORTANT = INVALID_USERNAME;
+
+    private final Environment environment;
+    private final String publicationsTableName;
+    private final Context context;
+    private final Instant mockNow = Instant.now();
+    private final Instant mockOneHourBefore = mockNow.minus(1, ChronoUnit.HOURS);
+    private DynamoDBDoiRequestsService doiRequestsService;
+    private UpdateDoiRequestHandler handler;
+
+
+    public ApiUpdateDoiRequestHandlerTest() {
+        environment = MockEnvironment.mockEnvironment();
+        publicationsTableName = environment.readEnv(ServiceConstants.PUBLICATIONS_TABLE_NAME_ENV_VARIABLE);
+        context = mock(Context.class);
+    }
+
+    @BeforeEach
+    public void init() {
+        initializeDatabase();
+        Clock mockClock = Clock.fixed(mockNow, ZoneId.systemDefault());
+        doiRequestsService = new DynamoDBDoiRequestsService(client, objectMapper, environment, mockClock);
+        handler = new UpdateDoiRequestHandler(environment, doiRequestsService);
+    }
+
+    @Test
+    public void handleRequestReturnsBadRequestWhenPublicationIdIsInvalid() throws IOException {
+        var updateDoiRequest = requestWithoutPublicationId();
+        updateDoiRequest.setPublicationId(INVALID_PUBLICATION_ID);
+
+        GatewayResponse<Problem> response =  sendRequest(updateDoiRequest, USERNAME_NOT_IMPORTANT);
+
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_BAD_REQUEST)));
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(details.getDetail(), containsString(INVALID_PUBLICATION_ID_ERROR));
+        assertThat(details.getDetail(), containsString(updateDoiRequest.getPublicationId()));
+    }
+
+    @Test
+    public void handleRequestReturnsBadRequestWhenNoPublicationId() throws IOException {
+        var updateDoiRequest = requestWithoutPublicationId();
+
+        GatewayResponse<Problem> response = sendRequest(updateDoiRequest, USERNAME_NOT_IMPORTANT,
+            Collections.emptyMap());
+
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_BAD_REQUEST)));
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(details.getDetail(), containsString(INVALID_PUBLICATION_ID_ERROR));
+        assertThat(details.getDetail(), containsString(NULL_STRING_REPRESENTATION));
+    }
+
+    @Test
+    public void handleRequestReturnsNotFoundWhenPublicationDoesNotExist() throws IOException {
+        var updateDoiRequest = new ApiUpdateDoiRequest();
+        var notExistingPublicationIdentifier = UUID.randomUUID().toString();
+        updateDoiRequest.setPublicationId(notExistingPublicationIdentifier);
+        updateDoiRequest.setDoiRequestStatus(DoiRequestStatus.APPROVED);
```

Comment:
Refactor lines 114-117 to method

Reply:
?

Reply:
Please explain what you are looking for here @axthosarouris

Reply:
the following 3 lines are repeated 6 times in the tests and they always do the same thing. 
refactor to 
```
    private ApiUpdateDoiRequest validUpdateRequest(String notExistingPublicationIdentifier) {
        var updateDoiRequest = new ApiUpdateDoiRequest();
        updateDoiRequest.setPublicationId(notExistingPublicationIdentifier);
        updateDoiRequest.setDoiRequestStatus(DoiRequestStatus.APPROVED);
        return updateDoiRequest;
    }
``` 
and call the method instead. One of the reasons is that when you do not have it in a method, one does not konw if the DoiRequestStatus.APPROVED plays an important role or not. By encapsulating it in a method you describe the intention with the name of the method

Reply:
Fixed, didnt see this actually. :)